src/components/drawers/MeasurementsDrawer/measurement-tables/MeasurementTableOptionsContainer.tsx

```diff
@@ -0,0 +1,197 @@
+import React, { useEffect, useMemo, useState } from "react";
+import { MeasurementTableOptions } from "./MeasurementTableOptions";
+import { SelectDialog } from "components/dialogs";
+import { Add } from "@mui/icons-material";
+import { batch, useDispatch, useSelector } from "react-redux";
+import { selectMeasurementTables } from "store/measurements/selectors";
+import {
+  selectCategoriesByKind,
+  selectKindDictionary,
+  selectThingsDictionary,
+} from "store/data/selectors";
+import { measurementsSlice } from "store/measurements/measurementsSlice";
+import { useDialogHotkey } from "hooks";
+import { DataArray } from "utils/file-io/types";
+import { HotkeyContext } from "utils/common/enums";
+
+export const MeasurementTableOptionsContainer = () => {
+  const tables = useSelector(selectMeasurementTables);
+  const categoriesByKind = useSelector(selectCategoriesByKind);
+  const kinds = useSelector(selectKindDictionary);
+  const [kindOptions, setKindOptions] = useState<string[]>(Object.keys(kinds));
+  const thingData = useSelector(selectThingsDictionary);
+  const dispatch = useDispatch();
+  const [expandedTable, setExpandedTable] = useState<string | undefined>();
+  const [loading, setLoading] = useState(false);
+  const worker: Worker = useMemo(
+    () => new Worker(new URL("./worker.ts", import.meta.url)),
+    []
+  );
+
+  const {
+    onClose: handleCloseTableDialog,
+    onOpen: handleOpenTableDialog,
+    open: isTableDialogOpen,
+  } = useDialogHotkey(HotkeyContext.ConfirmationDialog);
+
+  const handleCreateTable = (kind: string) => {
+    const thingIds = kinds[kind].containing;
+    const convertedThingData: {
+      id: string;
+      kind: string;
+      data: number[][][][];
+      encodedMask?: number[];
+      decodedMask?: DataArray;
+    }[] = thingIds.map((thingId) => {
+      const thing = thingData[thingId];
+      if ("encodedMask" in thing) {
+        return {
+          id: thing.id,
+          kind: kind,
+          data: thing.data.arraySync(),
+          encodedMask: thing.encodedMask,
+          decodedMask: thing.decodedMask,
+        };
+      } else {
+        return { id: thing.id, kind: kind, data: thing.data.arraySync() };
+      }
+    });
+
+    if (window.Worker) {
+      setLoading(true);
+      worker.postMessage({ kind: kind, things: convertedThingData });
+    }
+  };
+  const handleTableTreeExpand = (tableId: string) => {
+    setExpandedTable((currentId) => {
+      return tableId === currentId ? undefined : tableId;
+    });
+  };
+
+  useEffect(() => {
+    setKindOptions(Object.keys(kinds));
+  }, [kinds]);
+
+  useEffect(() => {
+    if (window.Worker) {
+      worker.onmessage = (
+        e: MessageEvent<{
+          kind: string;
+          data: Record<
+            string,
+            {
+              channels: number[][];
+              maskData: DataArray | undefined;
+              maskShape: { width: number; height: number } | undefined;
```

Comment:
This will eventually need to be done when measurements are calculated. So my thinking was we can do it up front, then not again. Or do it when the user selects a measurement. As I was working on it it seemed like the wait time was more sufferable when it was split up vs when it was done all together.

Reply:
Ah, my bad. For some reason I didn't see that it was being dispatched via `measurementsSlice.actions.updateMeasurements`, and therefore though it was just being discarded.